lib/config/configuration.go

```diff
@@ -937,8 +937,14 @@ func applyAppsConfig(fc *FileConfig, cfg *service.Config) error {
 			InsecureSkipVerify: application.InsecureSkipVerify,
 		}
 		if application.Rewrite != nil {
+			// Parse http rewrite headers if there are any.
+			headers, err := service.ParseHeaders(application.Rewrite.Headers)
```

Comment:
Is it deliberate to be permissive (letting the user shoot himself in the foot) with regard to the configured headers? Shouldn't we, e.g., warn the user about collisions in the supplied header names (considered as case-insensitive) or conflicts with Teleport-specific headers?

Reply:
Originally I thought yes, but after thinking some more decided it makes sense to not allow rewriting headers injected by Teleport. Updated.